homeassistant/components/tplink/common.py

```diff
@@ -0,0 +1,189 @@
+"""Common code for tplink."""
+
+import logging
+from datetime import timedelta
+from typing import Any, Callable, List
+
+from pyHS100 import SmartBulb, SmartDevice, SmartPlug, SmartDeviceException
+
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ATTR_CONFIG = 'config'
+CONF_DISCOVERY = 'discovery'
+CONF_LIGHT = 'light'
+CONF_SWITCH = 'switch'
+
+
+class SmartDevices:
+    """Hold different kinds of devices."""
+
+    def __init__(
+            self,
+            lights: List[SmartDevice] = None,
+            switches: List[SmartDevice] = None
+    ):
+        """Constructor."""
+        self._lights = lights or []
+        self._switches = switches or []
+
+    @property
+    def lights(self):
+        """Get the lights."""
+        return self._lights
+
+    @property
+    def switches(self):
+        """Get the switches."""
+        return self._switches
+
+    def has_device_with_host(self, host):
+        """Check if a devices exists with a specific host."""
+        for device in self.lights + self.switches:
+            if device.host == host:
+                return True
+
+        return False
+
+
+async def async_has_discoverable_devices(hass):
```

Comment:
This name is a bit weird, since I would expect `has_` to mean it returns a boolean, but here it is used to return actual devices. I would rename it to `async_discover_devices`